src/plugins/translate/TranslationManager.js

```diff
@@ -160,19 +153,22 @@ export class TranslationManager {
   }
 
   /**
-   * Targets the page and paragraph of a text layer to create a translation from the "fromLang" to the "toLang"
+   * Targets the page and paragraph of a text layer to create a translation from the "fromLang" to the "toLang". A priority can be 
    * @param {string} fromLang
    * @param {string} toLang
    * @param {string} pageIndex
    * @param {number} paragraphIndex
    * @param {string} text
+   * @param {number} [priority]
    * @return {Promise<string>} translated text
    */
 
-  getTranslation = async (fromLang, toLang, pageIndex, paragraphIndex, text) => {
+  getTranslation = async (fromLang, toLang, pageIndex, paragraphIndex, text, priority) => {
+    this.active = true;
     const key = `${fromLang}${toLang}-${pageIndex}:${paragraphIndex}`;
-
     const cachedEntry = this.alreadyTranslated.entries.find(x => x.index == key);
+    // Attempt to enforce ordered translation using priority pageIndex + paragraphIndex
+    const translationPriority = (priority ? priority : 0) + paragraphIndex;
```

Comment:
Oh because we do the math for priority in the caller of this method, it would make sense to do this math in the parent as well. Here we can just forward along the set priority.